File: src/commands/list.ts
import { loadStorage } from "../storage";

export function listItems() {
    const storage = loadStorage();

    if (Object.keys(storage).length === 0) {
        console.log("🔴 No stored items found.");
        return;
    }

    console.table(
        Object.entries(storage).map(([key, entry]) => ({
            Name: key,
            Type: entry.type,
        }))
    );
}

File: src/commands/list.rs
use crate::cloud::RemoteSession;
use crate::vault::load_vault;
use owo_colors::OwoColorize;

pub fn list_items(session: &RemoteSession) -> Result<(), String> {
    println!("{}", "Stored Items:".yellow().bold());

    let mut keys: Vec<String> = load_vault(session)?
        .into_iter()
        .map(|entry| entry.key)
        .collect();

    keys.sort();

    if keys.is_empty() {
        println!("{}", "No items found in your vault.".red());
    } else {
        for key in keys {
            println!("  - {}", key.bold().green());
        }
    }

    Ok(())
}
